homeassistant/components/camera/__init__.py

```diff
@@ -927,7 +928,11 @@ async def async_handle_snapshot_service(
             f"Cannot write `{snapshot_file}`, no access to path; `allowlist_external_dirs` may need to be adjusted in `configuration.yaml`"
         )
 
-    image = await camera.async_camera_image()
+    async with async_timeout.timeout(CAMERA_IMAGE_TIMEOUT):
```

Comment:
The existing code doesn't seem to use a timeout for snapshots

Reply:
Optional: You could also consider calling `_async_get_image` here to share timeout logic, but haven't thought this through.

Reply:
I'll take a look at that later, but I think it's probably better to avoid changing the PR more now.